From 40aa3851dd316bbd8ca965226275c28373f7f5bc Mon Sep 17 00:00:00 2001
From: Henry Li <lihl@microsoft.com>
Date: Fri, 28 Jul 2023 17:02:35 -0700
Subject: [PATCH] errors.Join is only supported in Golang v1.20+ and the latest
 version Mariner supports is 1.19.11. Given that the usage of errors.Join is
 only for logging purposes, currently remove the usage of it to build with
 Golang 1.19 and remove this patch once Mariner onbaords to Golang v1.20

---
 internal/runtime/logger.go | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/internal/runtime/logger.go b/internal/runtime/logger.go
index b3122ac..75a4be5 100644
--- a/internal/runtime/logger.go
+++ b/internal/runtime/logger.go
@@ -17,7 +17,6 @@
 package runtime
 
 import (
-	"errors"
 	"fmt"
 	"io"
 	"os"
@@ -62,7 +61,7 @@ func (l *Logger) Update(filename string, logLevel string, argv []string) {
 	if !configFromArgs.version {
 		configLogFile, err := createLogFile(filename)
 		if err != nil {
-			argLogFileError = errors.Join(argLogFileError, err)
+			argLogFileError = err
 		}
 		if configLogFile != nil {
 			logFiles = append(logFiles, configLogFile)
@@ -72,7 +71,7 @@ func (l *Logger) Update(filename string, logLevel string, argv []string) {
 		if argLogFile != nil {
 			logFiles = append(logFiles, argLogFile)
 		}
-		argLogFileError = errors.Join(argLogFileError, err)
+		argLogFileError = err
 	}
 	defer func() {
 		if argLogFileError != nil {
-- 
2.17.1

